npm/test/get-file-check-progress.test.ts

```diff
@@ -43,10 +40,7 @@ class GraphqlClientEmptyData {
     query: DocumentNode,
     variables: GetFileCheckProgressQueryVariables
   ) => Promise<FetchResult<GetFileCheckProgressQuery>> = async (_, __) => {
-    const data: GetFileCheckProgressQuery = {
-      getConsignment: null
-    }
-    return { data }
+    throw new Error("No progress metadata found for consignment")
```

Comment:
Is this the right test setup? The class is called `GraphqlClientEmptyData`.

Reply:
This is testing the case where [getConsignment](https://github.com/nationalarchives/tdr-transfer-frontend/blob/9273ee451f6652309835b6994757d5408d8a0ada/npm/src/filechecks/get-file-check-progress.ts#L55) is empty. Should I maybe rename it to `GraphqlClientEmptyGetConsignment`?

Reply:
Ah, I see. This case is now pretty much the same as `GraphqlClientFailure` then, because they both throw and `Error`. So I don't think you need two separate test cases any more.

Reply:
But they generate different errors though; in the case of a graphql client failure (this was the one where we removed the `catch` block), now it just throws an exception and in the case of empty data, we are throwing our own error with this message `No progress metadata found for consignment ${consignmentId}`. Should I still remove the test for the `GraphqlClientFailure`?

Reply:
After thinking about it and I came to the conclusion that it probably does make sense to have them separate, as this `GraphqlClientEmptyGetConsignment` test, checks that the user is seeing the message we expect them to see. If someone accidentally changed that message, the test would fail. Or if they purposely removed it, thinking that it's not necessary, the test failure might cause them to reconsider.

Also, if we merged the two failures and someone removed the `else` statement, we wouldn't know it as the test would still pass (since it would be just checking for an exception being thrown).

Reply:
Does the message get shown to the user regardless of what it says, or is there some logic that's depending on the wording of the message?

If it's the former, then that's what we should be testing - you wouldn't need multiple tests, just a test that whatever message is in the Error is show on screen.

Reply:
For the case of the graphql failure, it would show a different failure message than the empty data 'failure' but in the case of the empty data failure, the message would be the same each time except for the consignmentId, so I did a fuzzy match for just `No progress metadata found for consignment`.